### ISSUE TITLE
False positive for no-shadowed-variable with multiple index definitions

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#1258]
### Bug Report

- __TSLint version__: `3.8.1`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
interface IFoo {
    [i: string]: any;
    [i: number]: any;
}
```

with `tslint.json`:

```json
{
    "rules": {
        "no-shadowed-variable": true
    }
}
```

#### Actual behavior

(no-shadowed-variable) file.ts[6, 6]: shadowed variable: 'i'

#### Expected behavior

This should be fine, yes?

**Comment by chrismbarr** [comment]
why not just write it like this?

```typescript
interface IFoo {
    [i: string|number]: any;
}
```

**Comment by JoshuaKGoldberg** [comment]
@chrismbarr Not allowed before TS 1.5: https://github.com/Microsoft/TypeScript/issues/1715 

Otherwise the bug still holds for the following:

```typescript
interface IFoo {
    [i: string]: boolean;
    [i: number]: string|any;
}
```
